src/ua/lviv/iot/algo/part1/lab2/AbstractLaptopManager.java

```diff
@@ -0,0 +1,60 @@
+package ua.lviv.iot.algo.part1.lab2;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AbstractLaptopManager {
+    private final List<AbstractLaptop> laptopList = new LinkedList<>();
+
+    public void addLaptop(AbstractLaptop laptop) {
+        this.laptopList.add(laptop);
+    }
+
+    public List<AbstractLaptop> findLaptopWithRamValue(int value) {
+        return laptopList.stream().
+                filter(laptop -> laptop.getRam() == value).
+                collect(Collectors.toList());
+    }
+
+    public List<AbstractLaptop> findGamingLaptop() {
+        return laptopList.stream().
+                filter(laptop -> laptop.getClass().
+                        getSimpleName().
+                        equals("GamingLaptop")).
+                collect(Collectors.toList());
+    }
+
+    public static void main(String... args) {
+        AbstractLaptopManager laptopManager = new AbstractLaptopManager();
+        laptopManager.addLaptop(new Ultrabook("Dell XPS 13", 13.3f, 16, 2048, 12, 98, 1.2d, 14.8d));
+        laptopManager.addLaptop(new Ultrabook("HP Spectre x360", 13.3f, 16, 2048, 12, 97, 1.3d, 15.9d));
+        laptopManager.addLaptop(new GamingLaptop("ASUS ROG Zephyrus G14", 14f, 16, 1024, 10, 100, 2, "NVIDIA GeForce RTX 2060"));
+        laptopManager.addLaptop(new GamingLaptop("MSI GE75 Raider", 12f, 16, 1536, 5, 99, 2, "NVIDIA GeForce RTX 2070"));
+        laptopManager.addLaptop(new BusinessLaptop("Lenovo ThinkPad X1 Carbon Gen 9", 14f, 16, 512, 20, 96, 6, 3));
+        laptopManager.addLaptop(new BusinessLaptop("Dell Latitude 7420", 14f, 16, 512, 16, 95, 6, 3));
+        laptopManager.addLaptop(new NetBook("Asus VivoBook Flip 14 TP412FA", 14f, 8, 512, 9, 94, "Windows 10 Home", "HD"));
+        laptopManager.addLaptop(new NetBook("Lenovo IdeaPad Flex 5 14ARE05", 14f, 8, 512, 10, 93, "Windows 10 Home", "HD"));
+
+        for (AbstractLaptop laptop :
+                laptopManager.laptopList) {
+            System.out.println(laptop);
+        }
+
+        System.out.println("\n");
+
+        var laptops1 = laptopManager.findLaptopWithRamValue(8);
+        for (AbstractLaptop laptop :
+                laptops1) {
```

Comment:
```suggestion
        for (AbstractLaptop laptop : laptops1) {
```
I guess it's more of an intuitive thing when you should or should not split the line for better readability.  I chop the lines mostly when chain function calls, have lots of parameters, and wrap a line if it is just too long